File: pret/src/main/java/com/bank/pret/model/HistoStatus.java
package com.bank.pret.model;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * Représente l'historique des changements de statut d'un compte.
 * <p>
 * Champs :
 * <ul>
 *   <li>id : identifiant unique de l'historique</li>
 *   <li>dateChangement : date et heure du changement de statut</li>
 *   <li>compte : compte concerné par le changement</li>
 *   <li>status : nouveau statut appliqué</li>
 * </ul>
 */
@Entity
@Table(name = "histo_status")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class HistoStatus implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "date_changement", nullable = false)
    private LocalDateTime dateChangement;

    // 🔹 Relation vers Compte
    @ManyToOne(optional = false)
    @JoinColumn(name = "id_compte", nullable = false)
    private Compte compte;

    // 🔹 Relation vers Status
    @ManyToOne(optional = false)
    @JoinColumn(name = "id_status", nullable = false)
    private Status status;
}

File: depot/Models/HistoStatus.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace depot.Models
{
    [Table("histo_status")] 
    public class HistoStatus
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }
        
        [Required]
        [Column("date_changement")]
        public DateTime DateChangement { get; set; }
        
        // Foreign keys
        [Required]
        [Column("id_compte")]
        public int IdCompte { get; set; }
        
        [Required]
        [Column("id_status")]
        public int IdStatus { get; set; }
        
        // Navigation properties
        [ForeignKey("IdCompte")]
        public virtual Compte Compte { get; set; } = null!;
        
        [ForeignKey("IdStatus")]
        public virtual Status Status { get; set; } = null!;
    }
}